Can't build HEAD

```console
$ go version
go version go1.9.2 linux/amd64
```

I got following errors when build HEAD:
```console
$ make
🇩 bin/ctr
go install encoding: open /usr/local/go/pkg/linux_amd64_shared/encoding.a: permission denied
go install unicode/utf16: open /usr/local/go/pkg/linux_amd64_shared/unicode/utf16.a: permission denied
go install math/bits: mkdir /usr/local/go/pkg/linux_amd64_shared/math/: permission denied
go install context: open /usr/local/go/pkg/linux_amd64_shared/context.a: permission denied
go install crypto/subtle: mkdir /usr/local/go/pkg/linux_amd64_shared/crypto/: permission denied
go install crypto/internal/cipherhw: mkdir /usr/local/go/pkg/linux_amd64_shared/crypto: permission denied
go install internal/syscall/unix: mkdir /usr/local/go/pkg/linux_amd64_shared/internal/syscall/: permission denied
go install hash: open /usr/local/go/pkg/linux_amd64_shared/hash.a: permission denied
go install encoding/base64: mkdir /usr/local/go/pkg/linux_amd64_shared/encoding/: permission denied
go install sort: open /usr/local/go/pkg/linux_amd64_shared/sort.a: permission denied
go install log: open /usr/local/go/pkg/linux_amd64_shared/log.a: permission denied
go install internal/nettrace: open /usr/local/go/pkg/linux_amd64_shared/internal/nettrace.a: permission denied
go install container/list: mkdir /usr/local/go/pkg/linux_amd64_shared/container/: permission denied
go install math/rand: open /usr/local/go/pkg/linux_amd64_shared/math/rand.a: no such file or directory
go install internal/singleflight: open /usr/local/go/pkg/linux_amd64_shared/internal/singleflight.a: permission denied
go install crypto/rc4: open /usr/local/go/pkg/linux_amd64_shared/crypto/rc4.a: no such file or directory
go install encoding/binary: open /usr/local/go/pkg/linux_amd64_shared/encoding/binary.a: no such file or directory
go install bytes: open /usr/local/go/pkg/linux_amd64_shared/bytes.a: permission denied
go install vendor/golang_org/x/crypto/curve25519: mkdir /usr/local/go/pkg/linux_amd64_shared/vendor: permission denied
go install os/signal: mkdir /usr/local/go/pkg/linux_amd64_shared/os/: permission denied
go install strings: open /usr/local/go/pkg/linux_amd64_shared/strings.a: permission denied
```

I tried `v1.0.0`, which is fine.

We should revert that change. You shouldn't have to remove the default go installation to build containerd.

I gave it a try and things don't work anymore:

```
~/g/s/g/c/containerd ❯❯❯ sudo rm -rf /usr/local/go/pkg                                                                                                                                                                                                                     ⏎ master ✭ ✱ ◼
~/g/s/g/c/containerd ❯❯❯ make && sudo make install                                                                                                                                                                                                                         ⏎ master ✭ ✱ ◼
🇩 bin/ctr
go tool: no such tool "compile"
Makefile:143: recipe for target 'bin/ctr' failed
make: *** [bin/ctr] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -11,11 +11,9 @@ REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet 
 ifneq "$(strip $(shell command -v go 2>/dev/null))" ""
 	GOOS ?= $(shell go env GOOS)
 	GOARCH ?= $(shell go env GOARCH)
-	GOHOSTOS ?= $(shell go env GOHOSTOS)
 else
 	GOOS ?= $$GOOS
 	GOARCH ?= $$GOARCH
-	GOHOSTOS ?= $$GOHOSTOS
 endif
 
 WHALE = "🇩"
@@ -52,14 +50,6 @@ GO_BUILD_FLAGS=
 
 #include platform specific makefile
 -include Makefile.$(GOOS)
-
-# Add the "incremental" build option `-i` (installs the packages that are
-# dependencies of the target) to speed up future builds. Only for native builds,
-# otherwise it will try to install in the incorrect (not native) path and fail.
-# Note: this won't be necessary from 1.10 (https://tip.golang.org/doc/go1.10#build)
-ifeq ($(GOOS), $(GOHOSTOS))
-	GO_BUILD_FLAGS += -i
-endif
 
 # Flags passed to `go test`
 TESTFLAGS ?= -v $(TESTFLAGS_RACE)

